cfme/common/__init__.py

```diff
@@ -126,3 +119,50 @@ def validate_properties(self):
             assert(expected_value == shown_value,
                    'Property {} has wrong value, expected {} but was {}'
                    .format(property_tuple[1], expected_value, shown_value))
+
+
+class SummaryPage(object):
+    """
+    Middleware provider and other pages in middleware must extend this class
+    to get table details of `Properties`, `Relationships` and `Smart Management`
+    """
+
+    def properties_ui(self):
+        """
+        `Properties` table
+        Returns: dict value of table details
+        """
+        return self.summary_data_ui("Properties")
+
+    def relationships_ui(self):
+        """
+        `Relationships` table
+        Returns: dict value of table details
+        """
+        return self.summary_data_ui("Relationships")
+
+    def smart_management_ui(self):
+        """
+        `Smart Management` table
+        Returns: dict value of table details
+        """
+        return self.summary_data_ui("Smart Management")
+
+    def summary_data_ui(self, header):
```

Comment:
@jkrocil @psav @mfalesni I want to use some of the functions from outside provider page, for example in server/deployments pages. But some methods available only with providers(example: `get_detail`, `load_details`, `_on_detail_page`, etc.,).
I do not get a way to reuse those methods. Could you please advice?